src/dataServices/hybridDataService.ts

```diff
@@ -221,17 +221,34 @@ export class HybridDataService implements Persistence, BulkDataAccess {
     async deleteResource(request: DeleteResourceRequest) {
         this.assertValidTenancyMode(request.tenantId);
         const { resourceType, id, tenantId } = request;
-        const itemServiceResponse = await this.dbPersistenceService.readResource({ resourceType, id, tenantId });
-        const { versionId } = itemServiceResponse.resource.meta;
-        const { bulkDataLink } = itemServiceResponse.resource;
-        if (bulkDataLink) {
-            const [, deleteResponse] = await Promise.all([
-                S3ObjectStorageService.deleteObject(bulkDataLink),
+
+        const projectionExpression = 'id, meta, bulkDataLink';
+        const itemServiceResponses = await this.dbPersistenceService.readAllResourceVersions(
+            {
+                resourceType,
+                id,
+                tenantId,
+            },
+            projectionExpression,
+        );
+
+        const deleteRequests: Array<Promise<any>> = [];
+        itemServiceResponses.forEach((element: any) => {
```

Comment:
I would suggest to remove the items from the DDB first, and upon success delete them from the S3. Cause if deletion fails at DDB, we will stay with invalid object.
So far you don't check the result of deletion. Why do you wait it at all (see line 247)?

Reply:
I wait for them because by returning the response to the client. The deletion has to be completed.